core/jazz_events-handler/test/test.js

```diff
@@ -1,10 +1,263 @@
+              S: 'UPDATE_DEPLOYMENT'
+            }
+          }
+        ],
+        LastEvaluatedKey: "T3ST_1"
+      }
+
+      let payload2 = {
+        Items: [
+          {
+            EVENT_NAME: {
+              S: 'CREATE_BRANCH'
+            }
+          },
+          {
+            EVENT_NAME: {
+              S: 'UPDATE_BRANCH'
+            }
+          },
+          {
+            EVENT_NAME: {
+              S: 'DELETE_BRANCH'
+            }
+          }
+        ]
+      }
+      AWS.mock("DynamoDB.DocumentClient", "scan", (params, cb) => {
+        if (counter === 0) {
+          counter++;
+          return cb(null, payload1);
+        }
+        else {
+          return cb(null, payload2);
+        }
+      });
+      index.getEvents(config)
+        .then(res => {
+          expect(res).to.have.length(6);
+          AWS.restore("DynamoDB.DocumentClient");
+        });
+    });
+  });
+
+  describe('processEachEvent tests', function () {
+    beforeEach(() => {
+      let payload = {
+        "EVENT_ID": {
+          "S": "T3ST-1D"
+        },
+        "TIMESTAMP": {
+          "S": "2018-11-19T10:00:00:007"
+        },
+        "REQUEST_ID": {
+          "S": "T3ST-REQUEST-1D"
+        },
+        "EVENT_HANDLER": {
+          "S": "JENKINS"
+        },
+        "EVENT_NAME": {
+          "S": "CREATE_ASSET"
+        },
+        "SERVICE_NAME": {
+          "S": "jazztest-testapi"
+        },
+        "SERVICE_ID": {
+          "S": "T3ST-S3RV1C3-1D"
+        }
+      }
+      let encoded = Buffer.from(JSON.stringify(payload)).toString('base64');
+      event.Records[0].kinesis.data = encoded;
+      interestedEvents = [
+        'CREATE_DEPLOYMENT',
+        'CREATE_ASSET',
+        'UPDATE_DEPLOYMENT',
+        'CREATE_BRANCH',
+        'UPDATE_BRANCH',
+        'DELETE_BRANCH'
+      ]
+    });
+
+    it("should indicate error if DynamoDB.putItem fails", () => {
+      AWS.mock("DynamoDB.DocumentClient", "put", (params, cb) => {
+        return cb(err, null);
+      });
+      index.processEachEvent(event.Records[0], config, interestedEvents)
+        .catch(error => {
+          expect(error).to.deep.eq(err);
+          AWS.restore("DynamoDB.DocumentClient");
+        });
+    });
 
-describe('Sample', function() {
-  it('tests handler', function(done) {
+    it("should process interested event", () => {
+      AWS.mock("DynamoDB.DocumentClient", "put", (params, cb) => {
+        let data = {
+          "message": "success"
+        }
+        return cb(null, data);
+      });
+      let message = {
+        "message": "successfully processed interested event"
+      };
 
-    // Add your test cases here.
-    assert(true);
-    done();
+      index.processEachEvent(event.Records[0], config, interestedEvents)
+        .then(res => {
+          expect(res).to.deep.eq(message);
+          AWS.restore("DynamoDB.DocumentClient");
+        });
+    });
+
+    it("should not process un-interested event", () => {
```

Comment:
un-interesting

Reply:
left it as uninterested as the code refers to events as interested and uninterested.